controllers/tools/applications.js

```diff
@@ -0,0 +1,235 @@
+'use strict';
+const path = require('path');
+const express = require('express');
+const moment = require('moment');
+const concat = require('lodash/concat');
+const groupBy = require('lodash/groupBy');
+const maxBy = require('lodash/maxBy');
+const get = require('lodash/get');
+const mean = require('lodash/mean');
+const uniqBy = require('lodash/uniqBy');
+const minBy = require('lodash/minBy');
+const times = require('lodash/times');
+
+const { PendingApplication, SubmittedApplication } = require('../../db/models');
+const { getDateRange } = require('./helpers');
+
+const router = express.Router();
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+function applicationsByDay(responses) {
+    if (responses.length === 0) {
+        return [];
+    }
+
+    const grouped = groupBy(responses, function(response) {
+        return moment(response.createdAt).format(DATE_FORMAT);
+    });
+
+    const newestResponse = maxBy(responses, response => response.createdAt);
+    const oldestResponse = minBy(responses, response => response.createdAt);
+    const oldestResponseDate = moment(oldestResponse.createdAt);
+
+    const daysInRange = moment(newestResponse.createdAt)
+        .startOf('day')
+        .diff(oldestResponseDate.startOf('day'), 'days');
+
+    const dayData = times(daysInRange, function(n) {
+        const key = oldestResponseDate
+            .clone()
+            .add(n, 'days')
+            .format(DATE_FORMAT);
+
+        const responsesForDay = grouped[key] || [];
+
+        return {
+            x: key,
+            y: responsesForDay.length
+        };
+    });
+
+    return dayData;
+}
+
+function minMaxAvg(arr) {
+    const sorted = arr.slice().sort((a, b) => a - b);
+    return {
+        lowest: sorted[0] || 0,
+        highest: sorted[sorted.length - 1] || 0,
+        average: mean(sorted) || 0
+    };
+}
+
+function measureTimeTaken(data) {
+    const appDurations = data.map(row => {
+        const created = moment(row.startedAt);
+        const submitted = moment(row.createdAt);
+        return submitted.diff(created, 'days');
+    });
+    return minMaxAvg(appDurations);
+}
+
+function measureWordCounts(data) {
+    const wordCounts = data.map(
+        d =>
+            d.applicationSummary
+                .map(_ => _.value)
+                .join(' ')
+                .split(' ').length
```

Comment:
This is a bit of fun really as it doesn't capture every field (eg. the budget here gets output as `Object object`) but it's a rough approximation, which I like.

Reply:
Love this as long as we're clear that it's approximate.